python/data_structures/linked_list.py

```diff
@@ -30,10 +44,76 @@ def includes(self, search_value):
             current = current.next
         return False
 
+    def append(self, new_node_value):
+        '''
+        traverse through the list until the current.next === None.
+        set current.next to the new node.
+        ser the next of the new node to None.
+        '''
+        current = self.head
+        while current:
+            if current.next == None:
+                current.next = Node(new_node_value)
+                break
+            current = current.next
+
+    def insert_before(self, before, new_value):
+        '''
+        - traverse through the list until current.next == before
+        - if conidtion is met :
+        set current.next == Node(new_value, current.next)
+        - outside if current = current.next
+        '''
```

Comment:
Your docstrings should describe _what_ a method does, not _how_ it does it. Your docstring on line 96 for `insert_after` is actually a great example of being concise and descriptive.

I recommend changing the docstrings for `insert_before` and `append` to align more with what you have for `insert_after`.